index.bs

```diff
@@ -2643,6 +2687,32 @@ non-negative integer |defaultBudget|, and an [=aggregatable debug reporting conf
     with |value|, |budget|, |supportedTypes|, and |defaultConfig|.
 1. Return the [=tuple=] (|budget|, |config|).
 
+To <dfn>parse max event states</dfn> from a [=map=] |map| and
+a possibly null 32-bit positive integer |attributionScopeLimit|:
+1. If |attributionScopeLimit| is null:
+    1. If |map|["<code>[=source-registration JSON key/max_event_states=]</code>"] [=map/exists=], return an error.
+    1. Return null.
+1. Let |maxEventStates| be |map|["<code>[=source-registration JSON key/max_event_states=]</code>"].
+1. If |maxEventStates| is not an integer or is less than or equal to 0, return an error.
+1. If |maxEventStates| is greater than the user agent's [=max trigger-state cardinality=], return an error.
+1. Return |maxEventStates|.
+
+To <dfn>parse attribution scopes for source</dfn> from a [=map=] |map| and a possibly null 32-bit positive integer |attributionScopeLimit|:
+1. Let |result| be a new [=set=].
+1. If |attributionScopeLimit| is null:
+    1. If |map|["<code>[=source-registration JSON key/attribution_scopes=]</code>"] [=map/exists=], return an error.
+    1. Return |result|.
+1. If |map|["<code>[=source-registration JSON key/attribution_scopes=]</code>"] does not [=map/exist=], return |result|.
+1. Let |values| be |map|["<code>[=source-registration JSON key/attribution_scopes=]</code>"].
+1. If |values| is not a [=list=], return an error.
+1. [=list/iterate|For each=] |value| of |values|:
+    1. If |value| is not a [=string=], return an error.
+    1. If |value|'s [=string/length=] is greater than [=max length of attribution scope for source=], return an error.
+    1. [=set/Append=] |value| to |result|.
+1. If |result|'s [=set/size=] is greater than |attributionScopeLimit|, return an error.
+1. If |result|'s [=set/size=] is greater than [=max attribution scopes per source=], return an error.
```

Comment:
Let's check the size of `|value|` instead of `|result|` (i.e. prior to deduplication) so that implementations have more room for optimization.

IIUC that is what the header-validator changes in this PR are doing anyway.

Reply:
Actually, let's hold off while we discuss this further, though there is still the inconsistency between the spec and validator. (This is an existing issue for other fields, like `destination`.)

Reply:
@apasel422 please let me know we are ready to proceed here.

Reply:
I'm not sure what we should do. Feel free to leave it for now for consistency with the other set checks.

Reply:
thanks, resolving for now.